packages/core/database/src/migrations/internal-migrations/5.0.0-convert-identifiers-long-than-max-length.ts

```diff
@@ -0,0 +1,252 @@
+import type { Knex } from 'knex';
+import createDebug from 'debug';
+import fs from 'node:fs';
+import type { Migration } from '../common';
+import type { Metadata } from '../../metadata';
+import { type Database, type MetadataOptions } from '../..';
+import { getUnshortenedName } from '../../utils/identifiers/shortener';
+
+const debug = createDebug('strapi::database::migration');
+
+type NameDiff<T> = {
+  short: T;
+  full: T;
+};
+
+// TODO: key isn't really needed except for debugging, we can remove it
+type IdentifierDiffs = {
+  indexes: NameDiff<{ index: number; key: string; tableName: string; indexName: string }>[];
+  tables: NameDiff<{ index: number; key: string; tableName: string }>[];
+  columns: NameDiff<{ index: number; key: string; tableName: string; columnName: string }>[];
+};
+
+export const renameIdentifiersLongerThanMaxLength: Migration = {
+  name: '5.0.0-rename-identifiers-longer-than-max-length',
+  async up(knex, db) {
+    const md = db.metadata;
+    console.log('metadata', fs.writeFileSync('md.json', JSON.stringify(Array.from(md), null, 2)));
+
+    const maxLength = db.DEFAULT_MAX_IDENTIFIER_LENGTH;
+
+    if (maxLength === undefined) {
+      throw new Error('Could not retrieve maxLength from db configuration');
```

Comment:
is it possible for this to happen ? and if does the users are supposed to add the db config ?

Reply:
Nope, I just removed the option to change the max length since it was dangerous for no benefit and this got left in. Nice catch.